hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -460,6 +460,10 @@ private ExecutorService getCommandExecutor(
             LOG.error(gid + ": writeChunk writeStateMachineData failed: blockId"
                 + write.getBlockID() + " logIndex " + entryIndex + " chunkName "
                 + write.getChunkData().getChunkName() + e);
+            metrics.incNumWriteDataFails();
+            // write chunks go in parallel. It's possible that one write chunk
+            // see the stateMachine is marked unhealthy by other parallel thread.
+            stateMachineHealthy.set(false);
```

Comment:
So question; if a thread has marked the container as unhealthy why should a write be successful at all ?

Reply:
I know this patch is merged, but I have no way of understanding what this means -- so I appreciate some comments or feedback that explains what happens here?

Reply:
> So question; if a thread has marked the container as unhealthy why should a write be successful at all ?

If a container is marked unhealthy, write will marked fail and log Append wlll fail in ratis. This is juts incrementing fail count metrics here and marking the stateMachine for the pipeline unhealthy so that now new ratis snapshots can be taken .

Reply:
> I know this patch is merged, but I have no way of understanding what this means -- so I appreciate some comments or feedback that explains what happens here?

Updated the description to add clarity to this.

Reply:
How an unhealthy pipeline will be recovered? I got a lot of exception because the pipline is marked as unhealthy and remain in the unhealthy state...